src/compiler/types.rs

```diff
@@ -2411,6 +2912,100 @@ impl From<BaseInterfaceType> for InterfaceType {
     }
 }
 
+pub trait InterfaceTypeWithDeclaredMembersInterface {
```

Comment:
I think similar to `ResolvedType` `InterfaceTypeWithDeclaredMembers` is something that `InterfaceType`'s can "mutate into" at some point in their lifecycle, so again seems like could maybe use a typestate pattern or something but for now just always have `InterfaceType` implement `InterfaceTypeWithDeclaredMembers`